### DISCUSSION TITLE
Apply color gradient along the width of a linestring

### DISCUSSION THREAD

**Discussion by mlshapiro** [visgl/deck.gl#7302] [Q&A] [answered]
I've followed the conversation for [adding a gradient to Polygon layers](https://github.com/visgl/deck.gl/issues/2663).  This works well.

Is there any similar approach to add a gradient along the width of LineString?  If not, Is it possible to do this with a custom shader?

For example,

![rect325](https://user-images.githubusercontent.com/8190979/192800083-f6552778-480a-460a-a00e-1bf6ce6fdf80.png)

**Comment by AdriSolid** [comment, marked as answer]
[This codepen](https://codepen.io/AdriSolid/pen/gOzvvjR?editors=1010) can help
```javascript
class GradientLineLayer extends LineLayer {
  getShaders() {
    const shaders = super.getShaders();
    shaders.inject = {
      'fs:DECKGL_FILTER_COLOR': `\
        float y = geometry.uv.y;
        float absY = abs(geometry.uv.y);
        
        if (y < 0.0) {
          color = vec4(color.r, color.g, color.b, 1.0 - absY);
        } else {
          color = vec4(color.r, color.g, color.b, 1.0 - y);
        }
      `
    };
    return shaders;
  }
}
```

**Reply by mlshapiro** [reply]
wow amazing, thank you.

I am going to try and adapt this to work with line sublayer of the GeoJsonLayer class.

**Comment by Nitikaa** [comment]
Hi @AdriSolid can we also add animation effects on line using shaders ?
